lib/ansible/modules/cloud/amazon/iam_role_facts.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+
+DOCUMENTATION = '''
+---
+module: iam_role_facts
+short_description: Gather information on IAM roles
+description:
+    - Gathers information about IAM roles
+version_added: "2.5"
+requirements: [ boto3 ]
+author:
+    - "Will Thames (@willthames)"
+options:
+    name:
+        description:
+            - Name of a role to search for
+            - Mutually exclusive with C(prefix)
+        required: false
+        default: None
+        aliases:
+            - role_name
+    path_prefix:
+        description:
+            - Prefix of role I(path) to restrict IAM role search for
+            - Mutually exclusive with C(name)
+        required: false
+        default: None
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+# find all existing IAM roles
```

Comment:
Please use `-name: find all existing IAM roles` (and remove `-` from next line), same for other examples

Reply:
sure